src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int size;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private Node<K,V>[] tableNodes = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resizingTableNodes();
+        }
+        final int index = key == null ? 0 : key.getClass().hashCode() % capacity;
```

Comment:
```suggestion
        final int index = key == null ? 0 : Math.abs(key.hashCode()) % table.length;
```

Reply:
This check is available in several methods. It is necessary to put it in a private method, something like getIndex ()